content/en/docs/ops/common-problems/security-issues/index.md

```diff
@@ -408,3 +408,82 @@ second ensure that [keys and certificates are being delivered](#keys-and-certifi
+destination rule configuration to let client sidecar send mTLS or plaintext
+traffic correctly.
+
+If you still think there's an issue, follow the instructions below for
+investigation.
+
+1. Get the client envoy cluster configuration to ensure auto mTLS is configured.
+You should see `transportSocketMatches` configured for the given Envoy cluster.
+
+    {{< text bash >}}
+    $ POD=$(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name})
+    $ istioctl proxy-config clusters ${POD}.foo -ojson | grep 'name.*outbound.*httpbin' -A130
+    "name": "outbound|8000||httpbin.default.svc.cluster.local",
+    ...
+    "transportSocketMatches": [
+            {
+                "name": "tlsMode-istio",
+                "match": {
+                    "tlsMode": "istio"
+                },
+                "transportSocket": {
+                    "name": "envoy.transport_sockets.tls",
+                    "typedConfig": {
+                        "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext",
+                        "commonTlsContext": ...
+            }
+            {
+                "name": "tlsMode-disabled",
+                "match": {},
+                "transportSocket": {
+                    "name": "envoy.transport_sockets.raw_buffer"
+                }
+            }
+    {{< /text >}}
+
+1. Check the destination pod to see if the destination endpoint has the label
+`tlsMode` equals to `istio`.
+
+    {{< text bash >}}
+    $ POD=$(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name})
+    $ IP=$(kubectl get pod -lapp=sleep -o jsonpath="{.items[*].status.podIP}")
+    $ kubectl exec -it ${POD} -c istio-proxy -- curl 'localhost:15000/config_dump?include_eds=true' | grep ${IP} -A15 -B5
+      "endpoint": {
+        "address": {
+        "socket_address": {
+            "address": "10.32.28.136",
+            "port_value": 80
+        }
+        ...
+        "metadata": {
+          "filter_metadata": {
+            "envoy.transport_socket_match": {
+              "tlsMode": "istio"
+            }
+        },
+    {{< /text >}}
+
+1. Check Envoy upstream debug log to confirm what transport socket configuration is being
+used. This requires to change Envoy upstream log to debug level.
+
+    {{< text bash >}}
+    $ POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})
+    $ istioctl pc log sleep-8f795f47d-rfhdt --level debug
+    2021-04-08T22:35:22.650478Z     debug   envoy upstream  transport socket match, socket tlsMode-istio selected for host with address 10.32.28.136:80
+    {{< /text >}}
+
+    The log line above indicates that mTLS socket is selected for the connection to the
+    corresponding host. Please note that the connection creation might be done in advance
+    of the actual request, if you don't see such log line, try to recreate destination side
```

Comment:
Not sure how to `try to recreate destination side endpoint`

Reply:
My understanding based on the experiment: the connection pool caches some connection to the specific hosts in advance, therefore users might not see the log line of "transport socket xxx is selected" by the time they do `exec -- curl` from sleep pod. Recreating destination pod, would force client envoy to go through connection pool initialization logic again.

Reply:
Feel free to suggest some other wording to make it clearer.